tests/unittests/algo/hebo/test_hebo.py

```diff
@@ -0,0 +1,225 @@
+def skip_or_xfail_mark(
+    *args, reason: str, raises: type[Exception] | tuple[type[Exception], ...] = ()
+):
+    if raises and not RUN_QUICK:
+        return pytest.mark.xfail(*args, reason=reason, raises=raises)
+    return pytest.mark.skip(reason=reason)
+
+
+@pytest.fixture(params=_model_names)
+def model_name(request: SubRequest):
+    """Fixture that yields the `model_name` parameter to pass to the Hebo constructor."""
+    _model_name: ModelName = request.param
+    yield _model_name
+
+
+_poor_constructor = skip_or_xfail_mark(
+    reason="Constructor isn't well written (should pop stuff from kwargs)",
+    raises=TypeError,
+)
+_pymoo_error = skip_or_xfail_mark(
+    reason="causes an error in pymoo code (`self.sampling` is np.ndarray)",
+    raises=AttributeError,
+)
+_wrong_constructor_args = skip_or_xfail_mark(
+    reason="ES class isn't receiving required constructor arguments",
+    raises=TypeError,
+)
+_tournament_selection_bool_casting = skip_or_xfail_mark(
+    reason="ES does bool(np.ndarray) in pymoo/operators/selection/tournament.py",
+    raises=ValueError,
+)
+_numpy_broadcasting_issue = skip_or_xfail_mark(
+    reason="ES has a numpy broadcasting issue",
+    raises=ValueError,
+)
+evolutionary_strategy_marks: dict[EvolutionStrategyName, list[pytest.MarkDecorator]] = {
+    "nsga2": [],
+    "de": [_poor_constructor],
+    "brkga": [_poor_constructor],
+    "cmaes": [_pymoo_error],
+    "nelder": [_pymoo_error],
+    "nelder-mead": [_pymoo_error],
+    "pattern-search": [_pymoo_error],
+    "ctaea": [_wrong_constructor_args],
+    "moead": [_wrong_constructor_args],
+    "nsga3": [_wrong_constructor_args],
+    "rnsga2": [_wrong_constructor_args],
+    "rnsga3": [_wrong_constructor_args],
+    "unsga3": [_wrong_constructor_args],
+    "ga": [_tournament_selection_bool_casting],
+    "pso": [_numpy_broadcasting_issue],
+}
+default = skip_or_xfail_mark(reason=f"Isn't explicitly supported.")
+
+
+@pytest.fixture(
+    params=[
+        pytest.param(es, marks=evolutionary_strategy_marks.get(es, default))
+        for es in _es_names
+    ]
+)
+def evolutionary_strategy(request: SubRequest):
+    """Fixture that yields the `es` parameter to pass to the Hebo constructor."""
+    es_name: EvolutionStrategyName = request.param
+    yield es_name
+
+
+@pytest.fixture(autouse=True)
+def _config(model_name: ModelName, evolutionary_strategy: EvolutionStrategyName):
+    """Fixture that parametrizes the configuration used in the tests below."""
+    backup: HEBO.Parameters = TestHEBO.config["parameters"]
+    new_params: HEBO.Parameters = dataclasses.replace(
+        backup,
+        model_name=model_name,
+        evolutionary_strategy=evolutionary_strategy,
+    )
+    # Replace the value in the Config.
+    TestHEBO.config["parameters"] = new_params
+    yield
+    TestHEBO.config["parameters"] = backup
+
+
+@pytest.fixture(autouse=True)
+def xfail_if_unseeded_model_chosen(request: SubRequest):
```

Comment:
Can this be simplified following your PR improving the parametrization of algo test suite phases?

Reply:
(Could be a follow up PR)

Reply:
Fixed in 2a9d245fe1c404ed60cf859fbc23cf09d03a8e42